Microsoft.Identity.Web/StartupHelpers.cs

```diff
@@ -76,6 +76,9 @@ public static IServiceCollection AddAzureAdV2Authentication(this IServiceCollect
                 // and [Access Tokens](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens)
                 options.TokenValidationParameters.NameClaimType = "preferred_username";
 
+                // Force the account selection (to avoid automatic sign-in with the account signed-in with Windows)
+                options.Prompt = "select_account";
```

Comment:
I've added this prompt = select_account so that, when signing-out, the user is not automatically re-signed-in with the account with which they are signed-in in Windows (which could be disarable. We might want to provide an option to enable/disable this). There must have been a change (@kalyankrishna1 thinks it's been pushed by MSIT and been there for a long time on corpnet. I was only hit this week/today)

Reply:
as discussed, we get automatically signed in due to this experimental accelerated sign-in being tried by MSIT in MSFT corp tenant. If this was introduced, 99.99% of the users who only have a single user to sign-in will start seeing this prompt breaking their SSO experience. MSAL.js team tried this and had to scale back. 

---
In reply to: [299644164](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/121#discussion_r299644164) [](ancestors = 299644164)

Reply:
Fair enough @kalyankrishna1 , but for the 0.01% who want to sign-in with a different account (for instance their MSA), how can they do, given when they sign-out and click on sign-in again they are automatically signed-in with their corp account?
Or maybe we should only have the prompt=select_account when the user has pressed sign-out ?

Reply:
I understand this is frustrating, but this solution is more invasive than discussing and presenting a guidance for those developers who want a feature where account selection needs to be available. If this is applied to all web applications, then a user with say 5 application in 5 different tabs will have to select an account in each tab, including when their session times out after an hour or so. The is what happened with msal.js early adopters and they complained very heavily! Using private browser sessions is a pain, but I don't know what else to do..

---
In reply to: [299792295](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/121#discussion_r299792295) [](ancestors = 299792295)